src/components/html.cjsx

```diff
@@ -25,11 +25,28 @@ module.exports = React.createClass
     if html
       __html: html
 
-  componentDidMount: ->
+  componentDidMount:  -> @updateDOMNode()
+  componentDidUpdate: -> @updateDOMNode()
+
+  # Perform manipulation on HTML contained inside the components node.
+  updateDOMNode: ->
     # External links should open in a new window
     root = @getDOMNode()
     links = root.querySelectorAll('a')
     _.each links, (link) ->
       link.setAttribute('target', '_blank') unless link.getAttribute('href')?[0] is '#'
+
     # MathML should be rendered by MathJax (if available)
     window.MathJax?.Hub.Queue(['Typeset', MathJax.Hub, root])
+    # Once MathML finishes processing, manually cleanup after it to prevent
+    # React "Invariant Violation" exceptions.
+    # MathJax calls Queued events in order, so this should always execute after typesetting
+    window.MathJax?.Hub.Queue([ ->
+      for nodeId in ['MathJax_Message', 'MathJax_Hidden', 'MathJax_Font_Test']
```

Comment:
I wonder if `document.querySelectorAll(['#MathJax_Message', '#MathJax_Hidden', '#MathJax_Font_Test'].join(','))` would be quicker